mantle/kola/tests/misc/network.go

```diff
@@ -293,35 +363,141 @@ var (
+					"enabled": true,
+					"name": "capture-macs.service"
+				},
+				{
+					"contents": "[Unit]\nDescription=Create VETH Pairs and Configue OVS interface over bond\nAfter=network-online.target\nAfter=capture-macs.service\nAfter=openvswitch.service\n\n[Service]\nType=oneshot\nExecStart=/usr/local/bin/create-veth-pairs\n\n[Install]\nRequiredBy=multi-user.target\n",
+					"enabled": true,
+					"name": "create-veth-pairs.service"
+				}
+			]
+		}
+	}`,
+		base64.StdEncoding.EncodeToString([]byte(defaultLinkConfig)),
+		base64.StdEncoding.EncodeToString([]byte(dhcpClientConfig)),
+		base64.StdEncoding.EncodeToString([]byte(captureMacsScript)),
+		base64.StdEncoding.EncodeToString([]byte(setupVethPairs)),
+		base64.StdEncoding.EncodeToString([]byte(setupOvsScript))))
+
+	switch pc := c.Cluster.(type) {
+	// These cases have to be separated because when put together to the same case statement
+	// the golang compiler no longer checks that the individual types in the case have the
+	// NewMachineWithQemuOptions function, but rather whether platform.Cluster
+	// does which fails
+	case *unprivqemu.Cluster:
+		m, err = pc.NewMachineWithQemuOptions(userdata, options)
+	default:
+		panic("unreachable")
+	}
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Add karg needed for the ignition to configure the network properly.
+	addKernelArgs(c, m, []string{fmt.Sprintf("macAddressList=%s,%s", primaryMac, secondaryMac)})
+}
+
 // NetworkSecondaryNics verifies that secondary NICs are created on the node
 func NetworkSecondaryNics(c cluster.TestCluster) {
 	primaryMac := "52:55:00:d1:56:00"
 	secondaryMac := "52:55:00:d1:56:01"
 
 	setupMultipleNetworkTest(c, primaryMac, secondaryMac)
 
-	checkOvsBridge(c, primaryMac)
+	checkOvsBridge(c, primaryMac, "")
 }
 
-func checkOvsBridge(c cluster.TestCluster, setPrimaryMac string) {
+func checkOvsBridge(c cluster.TestCluster, setPrimaryMac, setPrimaryIp string) {
 	var err error
-	machineIdx := 0
-
+	m := c.Machines()[0]
 	ovsBridgeName := "brcnv-iface"
-	bondInterfaceMACAddress, err := getBondIfaceMAC(c, machineIdx, ovsBridgeName)
+	bondInterfaceMACAddress, err := getBondIfaceMAC(c, m, ovsBridgeName)
 	if err != nil {
 		c.Fatalf("failed to fetch bond MAC Address: %v", err)
 	}
 
 	// check bond interface got the primary MAC
 	if bondInterfaceMACAddress != setPrimaryMac {
-		c.Fatalf("ovs-bridge %s primary MAC %s does not match expected IP", ovsBridgeName, bondInterfaceMACAddress, setPrimaryMac)
+		c.Fatalf("ovs-bridge %s primary MAC %s does not match expected MAC %s", ovsBridgeName, bondInterfaceMACAddress, setPrimaryMac)
 	}
-}
 
-func getBondIfaceMAC(c cluster.TestCluster, machineIdx int, interfaceName string) (string, error) {
-	m := c.Machines()[machineIdx]
+	if setPrimaryIp != "" {
```

Comment:
Can the content of this be moved to a separate function, so we don't bloat this one?

Reply:
moved each to a functions to avoid bloating. Done